### PR TITLE
[NAN-851] connection syncs improvements

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Sync connection page improvements. 
Use consistent method to get the schedule. Remove schedule discrepancy logic

## Connection Page Improvements:
- Should be inactive while waiting for a success/failure response 
- Should show a loading state while waiting for a success/failure response 
- Should show a toast in all cases (success & failure) 
- The page should update the sync table accordingly (without having to refresh) — page refreshes every 10 seconds
- if triggering job: update to running THEN completed/failed accordingly (ideally without refreshing)

# Bug Fixes
* Sync connection UI shows a sync is paused when it shouldn't be
* Next sync time is often missing

We were using the `listSchedules` which didn't give reliable schedule information. I noticed when testing that the API sync status endpoint gave back reliable information when the UI didn't so the UI now uses the same temporal API of `describeSchedule` to reliably get the schedule state and the future run times.

## Issue ticket number and link
NAN-851

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/sync/sync.service.ts [hunk 1]
```diff
@@ -283,26 +284,14 @@ export const getSyncs = async (nangoConnection: Connection): Promise<(Sync & { s
...
-        } else if (!schedule?.info?.paused && sync.schedule_status === SyncStatus.PAUSED) {
-            sync = {
-                ...sync,
-                schedule_status: SyncStatus.RUNNING
-            };
-            await updateScheduleStatus(schedule_id, SyncCommand.UNPAUSE, null, nangoConnection.environment_id);
+        const schedule = await syncClient?.describeSchedule(schedule_id);
+        let futureActionTimes: number[] = [];
+        if (schedule) {
+            futureActionTimes = schedule?.info?.futureActionTimes?.map((long) => long.seconds?.toNumber()) as number[];
```
Comment (TBonnin) [inline packages/shared/lib/services/sync/sync.service.ts]: would the value of `futureActionTimes` be `undefined` if `.info` or `.futureActionTimes` is undefined?

File: packages/shared/lib/services/sync/sync.service.ts [hunk 2]
```diff
@@ -283,26 +284,14 @@ export const getSyncs = async (nangoConnection: Connection): Promise<(Sync & { s
...
-                schedule_status: SyncStatus.PAUSED
-            };
-            await updateScheduleStatus(schedule_id, SyncCommand.PAUSE, null, nangoConnection.environment_id);
-        } else if (!schedule?.info?.paused && sync.schedule_status === SyncStatus.PAUSED) {
-            sync = {
-                ...sync,
-                schedule_status: SyncStatus.RUNNING
-            };
-            await updateScheduleStatus(schedule_id, SyncCommand.UNPAUSE, null, nangoConnection.environment_id);
+        const schedule = await syncClient?.describeSchedule(schedule_id);
```
Comment (TBonnin) [inline packages/shared/lib/services/sync/sync.service.ts:290]: Make me think our scheduler is going to need this functionality :)

File: packages/webapp/src/pages/Connection/Show.tsx [hunk 3]
```diff
@@ -151,7 +141,7 @@ We could not retrieve and/or refresh your access token due to the following erro
         return <PageNotFound />;
     }
 
-    if (!loaded || !syncLoaded) {
+    if (!loaded || syncLoading) {
```
Comment (TBonnin) [inline packages/webapp/src/pages/Connection/Show.tsx:144]: wha's the difference between `!loaded` and `syncLoading`

**Review by TBonnin — APPROVED** [review]

File: packages/webapp/src/pages/Connection/Show.tsx (hunk 3 above)
Comment (khaliqgant) [reply packages/webapp/src/pages/Connection/Show.tsx:144]: Loaded is the page frame and meta information like the connection and provider and syncLoading is the actual sync data

File: packages/shared/lib/services/sync/sync.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/sync/sync.service.ts:290]: Yup, indeed

File: packages/shared/lib/services/sync/sync.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/services/sync/sync.service.ts]: Updated this to check for these properties.

Comment (TBonnin) [reply packages/shared/lib/services/sync/sync.service.ts]: you can just default to `|| []` so you don't have to check

Comment (khaliqgant) [reply packages/shared/lib/services/sync/sync.service.ts]: Linter complains when I try to update to this, going to leave as is for now -- unless you have any issues

**Pull request merged**